core/src/test/java/org/mule/routing/correlation/EventCorrelatorTestCase.java

```diff
@@ -70,6 +72,8 @@
     @Mock(answer = Answers.RETURNS_DEEP_STUBS)
     private ListableObjectStore mockProcessedGroups;
     @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private Lock mockLock;
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
```

Comment:
use an actual lock implementation rather than a mock. Otherwise some race conditions may occur that cause flakiness in this test

Reply:
I've changed the logic of the creation of the lock. So now it gets the lock internally in the EventProcessor from the factory in the muleContext. No need to pass it as a parameter in the constructor.